src/plugins/lightscript.js

```diff
@@ -405,16 +424,24 @@ pp.parseIf = function (node, isExpression) {
     node.consequent = this.parseStatement(false);
   }
 
-  node.alternate = this.parseIfAlternate(node, isExpression, indentLevel);
+  node.alternate = this.parseIfAlternate(node, isExpression, isColon, indentLevel);
+
+  if (isColon) this.popBlockState();
 
   return this.finishNode(node, isExpression ? "IfExpression" : "IfStatement");
 };
 
-pp.parseIfAlternate = function (node, isExpression, indentLevel) {
-  const isColon = node.consequent.extra && node.consequent.extra.curly === false;
-
-  // if whitespace-block, must match indent level
-  if (isColon && this.state.indentLevel !== indentLevel) {
+pp.parseIfAlternate = function (node, isExpression, ifIsWhiteBlock, ifIndentLevel) {
+  if (!this.match(tt._elif) && !this.match(tt._else)) return null;
+
+  // If the indent level here doesn't match with the current whiteblock `if`, or
+  // it matches with a whiteblock `if` higher on the stack, then this alternate
+  // clause does not match the current `if` -- so unwind the recursive descent.
+  const alternateIndentLevel = this.state.indentLevel;
+  if (
+    ( alternateIndentLevel !== ifIndentLevel ) &&
```

Comment:
I try to avoid style nits, but the extra spaces here don't match (though I personally prefer them as well)